service/src/main/java/bio/terra/workspace/common/utils/ControllerUtils.java

```diff
@@ -35,5 +38,15 @@ public static String getAsyncResultEndpoint(HttpServletRequest request, String j
     return getAsyncResultEndpoint(request, jobId, "result");
   }
 
+  /**
+   * Return the appropriate response code for an endpoint, given an async job report. For a job
+   * that's still running, this is 202. For a job that's finished (either succeeded or failed), the
```

Comment:
I thought we couldn't return two distinct 2xx codes from an endpoint (like I was trying to do with 200 and 204).

Reply:
Right, let me be specific. You cannot generate two distinct 2xx codes that return _different_ responses.
The code gen will pick one to be the type returned by the client.

If you check, we have a bug in the current `createCloudContext` where the 202 thinks it is returning something different from the 200. But when you generate the client code for it, the signature is:
```
public CreateCloudContextResult createCloudContext(...
```
returning the 200 result. Apparently, whatever tests we have do not look at the 202 result. Or perhaps because we finish the flight fast enough that we never have a wait (?! seems unlikely).

@zloery what do you think about fixing that as part of this PR?

Reply:
Double checking that I understand this bug correctly: our yaml has different return types for 200 vs 202, but really only returns a `CreateCloudContextResultResponse` (the 200 return type)?

From what I can tell, the code generated from this yaml only ever uses the 200 return type. The swagger-ui and client library both use the 200 return type in all cases (even as the response type for 202 in swagger-ui). Without digging too deep, this seems like a  quirk of codegen: if you give it multiple return types it just picks one (the first? lowest code? I'm not sure) and uses that everywhere.

Since the 202 return type is ignored everywhere, I've fixed it in the yaml without needing to change other places. The other async endpoints (and their result endpoints) all match 200 and 202 responses.